iFixit/react-commerce
# Encode Canonical URL

## Overview

We've had some errors like [this](https://sentry.io/organizations/ifixit/issues/3504489498/events/d9719aa880c2447abada45d57cdcf3fe/?project=6469069) where there seems to be some sort of link that is not encoded correctly. This bug could potentially come from our canonical url not being encoded, so this PR just makes sure that we encode the url correctly.

## CR
is there any harm in url encoding these urls?

## QA
Make sure that the links still work I guess. 
Check on David's comment: https://github.com/iFixit/react-commerce/pull/654#issuecomment-1235663872

Connects #632
